src/main/java/com/google/devtools/build/lib/analysis/RunfilesSupport.java

```diff
@@ -495,4 +524,83 @@ public static Path inputManifestPath(Path runfilesDir) {
   public static Path outputManifestPath(Path runfilesDir) {
     return runfilesDir.getRelative(OUTPUT_MANIFEST_BASENAME);
   }
+
+  @Nullable
+  private static Artifact createRepoMappingManifestAction(
+      RuleContext ruleContext, Runfiles runfiles, Artifact owningExecutable) {
+    NestedSet<Package> transitivePackages =
+        ruleContext.getTransitivePackagesForRunfileRepoMappingManifest();
+    if (transitivePackages == null) {
+      // For environments where transitive packages are not tracked, we don't have external repos,
+      // so don't build the repo mapping manifest in such cases.
+      return null;
+    }
+
+    ImmutableList<Entry> entries = collectRepoMappings(transitivePackages, runfiles);
+    if (entries == null) {
+      // If Bzlmod is not enabled, nobody reads the repo mapping manifest anyway, so don't generate
```

Comment:
Can this branch be taken even if Bzlmod is enabled, for example in repositories defined by `WORKSPACE`?

Reply:
Reading the definition of `collectRepoMappings` answered that question.

Reply:
Fixed the comment.